### PR TITLE
JAPI-262 Remove cluster param from wsdfu::deletefile

### PR DESCRIPTION
**Pull request by rpastrana**
Signed-off-by: Pastrana, Rodrigo <rodrigo.pastrana@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [ ] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Comment by rpastrana** [comment]
WSDFU method does not provide "cluster" parameter option, user must use filename@group notation for cluster specific actions

**Comment by rpastrana** [comment]
@jpmcmu please review

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java [hunk 1]
```diff
@@ -873,14 +873,14 @@ public void deleteSuperFileSubfiles(String superfile, List<String> subfiles) thr
...
      * @param files
-     *            - list of filenames to delete
+     *            - list of filenames to delete - use "filename@group" notation for cluster specific files
      * @param cluster
      *            - name of cluster to delete from (will delete from all clusters if null)
      * @return list of results of file deletions
      * @throws Exception
      * @throws ArrayOfEspExceptionWrapper
      */
-    public List<DFUResultWrapper> deleteFiles(Set<String> files, String cluster) throws Exception, ArrayOfEspExceptionWrapper
```
[the remark this answers is not in the document]
Comment (lrenn) [reply wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java]: Might want to mark deprecated, add the new signature (old one calls new one ignoring param).  Removed deprecated in next release.

**Comment by rpastrana** [comment]
@jpmcmu please review

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java [hunk 2]
```diff
@@ -888,7 +890,26 @@ public void deleteSuperFileSubfiles(String superfile, List<String> subfiles) thr
...
+
+        if (cluster != null && !cluster.isEmpty())
+        {
+            for(int i = 0; i < filesarray.length; i++)
+            {
+                String fullfile = filesarray[i];
+                String[] split = fullfile.split("::");
+                String file = split[split.length-1];
+
+                int atIndex = file.indexOf('@');
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java]: You probably already mentioned this, but can files have the '@' symbol in their name? If so would this create false positives for those files?

Comment (rpastrana) [reply wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java]: Good question, but from documentation and a few tests, the @ symbol can only be used as a delimiter between filename and groupname. the odd thing is that the scopes can have the @ symbol

Comment (jpmcmu) [reply wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java]: Gotcha looks good to me

**Pull request merged**